geonode/upload/models.py

```diff
@@ -102,7 +102,7 @@ class Upload(models.Model):
     layer = models.ForeignKey(Layer, null=True, on_delete=models.SET_NULL)
     upload_dir = models.TextField(null=True)
     store_spatial_files = models.BooleanField(default=True)
-    name = models.CharField(max_length=64, null=True)
```

Comment:
Should we add in the migration file, some `RunPython` in order to generate a random unique name where this value is null?

Reply:
@afabiani as an instance, I just tried to run the migration on my local instance and it raise the duplicate key error:
```python
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UniqueViolation: could not create unique index "upload_upload_user_id_name_7c9d1782_uniq"
DETAIL:  Key (user_id, name)=(1000, road_network) is duplicated.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 29, in <module>
    execute_from_command_line(sys.argv)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/core/management/base.py", line 83, in wrapped
    res = handle_func(*args, **kwargs)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 232, in handle
    post_migrate_state = executor.migrate(
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/migrations/executor.py", line 245, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/migrations/migration.py", line 124, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/migrations/operations/models.py", line 527, in database_forwards
    alter_together(
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 367, in alter_unique_together
    self.execute(self._create_unique_sql(model, columns))
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/backends/base/schema.py", line 137, in execute
    cursor.execute(sql, params)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/backends/utils.py", line 99, in execute
    return super().execute(sql, params)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/mattia/.virtualenvs/geonode33x/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.IntegrityError: could not create unique index "upload_upload_user_id_name_7c9d1782_uniq"
DETAIL:  Key (user_id, name)=(1000, road_network) is duplicated.
```